src/main/scala/nl/knaw/dans/lib/logging/DebugEnhancedLogging.scala

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright (C) 2016 DANS - Data Archiving and Networked Services (info@dans.knaw.nl)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package nl.knaw.dans.lib.logging
+
+import com.typesafe.scalalogging.LazyLogging
+
+trait DebugEnhancedLogging extends LazyLogging {
+  /**
+   * Logs a trace level statement with the function name, parameter names and parameter values.
+   *
+   * Example:
+   * {{{
+   *   // Function definition
+   *   def someFunction(s: String, i: Int) = {
+   *    trace(s, i)
+   *    // Your code ...
+   *   }
+   *
+   *   // Function call
+   *   someFunction("My input string", 2)
+   *
+   *   // This will result in a logging line like this (if the TRACE log level is enabled):
+   *   [TRACE] someFunction [(s, i)]:  ("My input string",2)
+   *
+   * }}}
+   *
+   * @param value list of parameter values
+   * @param enclosing implicitly passed in context from the sourcecode library
+   * @tparam V type parameter, no idea how it works
+   */
+  def trace[V](value: sourcecode.Text[V])(implicit enclosing: sourcecode.Name) = {
+    logger.trace(s"${enclosing.value} [${value.source}]: ${value.value}")
+  }
+
+
+  /**
+   * Logs a custom debug level statement with the calling function name prefixed
+   *
+   * Example:
+   * {{{
+   *   // Function definition
+   *   def someFunction(s: String, i: Int) = {
+   *    // Your code ...
+   *    debug("Doing important stuff here")
+   *   }
+   *
+   *
+   *   // This will result in a logging line like this (if the DEBUG log level is enabled):
+   *   [DEBUG] someFunction: Doing important stuff here
+   *
+   * }}}
+   * @param s
+   * @param enclosing
+   */
+  def debug(s: String)(implicit enclosing: sourcecode.Name) = logger.debug(s"${enclosing.value}: $s")
```

Comment:
With an [implicit class](https://github.com/DANS-KNAW/dans-scala-lib/pull/1/files/48f3702f7abbc5a9d21008d413b59b7a7aa2dc74#diff-bc7d96adb19a6c6f7072e2d299516c97R20) you can extend third party classes. 

My point:
When importing `logger._` in advance I don't have the help of auto-completion. My habit is: type the object, use auto-completion, cursor on the method, alt-enter, import static. Using auto-completion I need less to remember from the top of my head and without trying to remember the chance increases I see new improved methods to adopt.